pkg/client/testclient/fake.go

```diff
@@ -38,6 +41,9 @@ func NewSimpleFake(objects ...runtime.Object) *Fake {
 // Invokes registers the passed fake action and reacts on it if a ReactFn
 // has been defined
 func (c *Fake) Invokes(action FakeAction, obj runtime.Object) (runtime.Object, error) {
+	c.Lock()
+	defer c.Unlock()
+
 	c.Actions = append(c.Actions, action)
 	if c.ReactFn != nil {
 		return c.ReactFn(ktestclient.FakeAction(action))
```

Comment:
How can the reactfn use the client (`type ReactionFunc func(FakeAction) (runtime.Object, error)`)? If it uses it - maybe via returning a closure? - how will this deadlock? Even if you run `Invokes` inside `Invokes` you wouldn't deadlock since deferred functions run before returns, right? You would have to run a closure with another Invokes inside to deadlock this. Anyway, there is already a PR upstream fixing this: https://github.com/GoogleCloudPlatform/kubernetes/pull/10783/